src/TestSuite/IntegrationTestCase.php

```diff
@@ -424,8 +424,7 @@ protected function _buildRequest($url, $method, $data)
     protected function _addTokens($url, $data)
     {
         if ($this->_securityToken === true) {
-            $keys = Hash::flatten($data);
-            $tokenData = $this->_buildFieldToken($url, array_keys($keys));
+            $tokenData = $this->_buildFieldToken($url, array_keys($data));
```

Comment:
I'm not sure this will work for deeply nested data sets. The scenarios I had in mind were things like inputs using `comments.0.author_id` as a field name.